File: src/main/kotlin/com/forestFire/ihbar/entity/FireReport.kt
package com.forestFire.ihbar.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "fire_reports")
data class FireReport(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(nullable = false)
    val latitude: Double,

    @Column(nullable = false)
    val longitude: Double,

    @Column(columnDefinition = "TEXT")
    val description: String? = null,

    @Column(name = "photo_url")
    val photoUrl: String? = null,

    @Column(name = "reporter_phone")
    val reporterPhone: String? = null,

    @Column(name = "reported_at", nullable = false)
    val reportedAt: LocalDateTime = LocalDateTime.now(),

    @Enumerated(EnumType.STRING)
    val status: ReportStatus = ReportStatus.PENDING,

    @Column(name = "address")
    val address: String? = null,

    @Column(name = "urgency_level")
    val urgencyLevel: UrgencyLevel = UrgencyLevel.MEDIUM
)

enum class ReportStatus {
    PENDING, VERIFIED, IN_PROGRESS, RESOLVED, FALSE_ALARM
}

enum class UrgencyLevel {
    LOW, MEDIUM, HIGH, CRITICAL
}

File: frontend/src/types/FireReport.ts
export interface FireReportRequest {
    latitude: number;
    longitude: number;
    description?: string;
    reporterPhone?: string;
    urgencyLevel: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
}

export interface FireReportResponse {
    success: boolean;
    message: string;
    reportId: number | null;
    data?: {
        id: number;
        latitude: number;
        longitude: number;
        description: string | null;
        photoUrl: string | null;
        reporterPhone: string | null;
        reportedAt: string;
        status: string;
        address: string | null;
        urgencyLevel: string;
    };
}

export interface FireReportData {
    id: number;
    latitude: number;
    longitude: number;
    description: string | null;
    photoUrl: string | null;
    reporterPhone: string | null;
    reportedAt: string;
    status: string;
    address: string | null;
    urgencyLevel: string;
}
